components/carousel.tsx

```diff
@@ -10,27 +10,29 @@ export async function Carousel() {
 
   return (
     <div className=" w-full overflow-x-auto pb-6 pt-1">
-      <div className="flex animate-carousel gap-4">
+      <ul className="flex animate-carousel gap-4">
         {[...products, ...products].map((product, i) => (
-          <Link
-            key={`${product.handle}${i}`}
-            href={`/product/${product.handle}`}
-            className="h-[30vh] w-2/3 flex-none md:w-1/3"
-          >
-            <GridTileImage
-              alt={product.title}
-              label={{
-                title: product.title,
-                amount: product.priceRange.maxVariantPrice.amount,
-                currencyCode: product.priceRange.maxVariantPrice.currencyCode
-              }}
-              src={product.featuredImage?.url}
-              width={600}
-              height={600}
-            />
-          </Link>
+          <li className="relative aspect-square h-[30vh] max-h-[275px] w-2/3 max-w-[475px] flex-none md:w-1/3">
+            <Link
+              key={`${product.handle}${i}`}
+              href={`/product/${product.handle}`}
+              className="relative h-full w-full"
+            >
+              <GridTileImage
+                alt={product.title}
+                label={{
+                  title: product.title,
+                  amount: product.priceRange.maxVariantPrice.amount,
+                  currencyCode: product.priceRange.maxVariantPrice.currencyCode
+                }}
+                src={product.featuredImage?.url}
+                fill
+                sizes="(min-width: 1024px) 16vw, (min-width: 768px) 25vw, 50vw"
```

Comment:
This is `home page → scrolling product carousel`.

Reply:
How did you come up with 16vw? Are there 6 columns at 1024px?

Reply:
This is a matter of personal preference, but generally I like using `max-width` so the left most is the smallest device and each device is a larger screen as you go to the right, and the fallback is for screen sizes that are super wide.

For example, see the code snippet here: https://nextjs.org/docs/pages/api-reference/components/image#sizes

Reply:
> How did you come up with 16vw? Are there 6 columns at 1024px?

I honestly don't know what I was thinking... I just pushed a commit to do 50vw → 33vw → 25vw.

Reply:
> This is a matter of personal preference, but generally I like using `max-width` so the left most is the smallest device and each device is a larger screen as you go to the right

I can totally do it this way if that is the preference / best practice. This just fit my mental model paired with Tailwind as it operates with `min-width`. Made matching the surrounding styles with `md` and `lg` easier for my brain. 

https://tailwindcss.com/docs/responsive-design